core/txpool/legacypool/legacypool.go

```diff
@@ -1739,9 +1742,26 @@ func (pool *LegacyPool) demoteUnexecutables() {
 			// Internal shuffle shouldn't touch the lookup set.
 			pool.enqueueTx(hash, tx, false, false)
 		}
-		pendingGauge.Dec(int64(len(olds) + len(drops) + len(invalids)))
+
+		conditionalDrops, err := list.FilterTransactionConditionals(pool.currentState)
+		if err != nil {
+			log.Trace("%d stale transactions dropped with failed conditional: %w", len(conditionalDrops), err)
+		}
+
+		txConditionalDemotedCounter.Inc(int64(len(conditionalDrops)))
+		for _, tx := range conditionalDrops {
+			if conditional := tx.Conditional(); conditional != nil { // not nil but just in case
+				txConditionalDemotedTimer.UpdateSince(conditional.SubmissionTime)
+			}
+
+			hash := tx.Hash()
+			pool.all.Remove(tx.Hash())
```

Comment:
The `tx.Hash()` is cached inside of the tx object, but since it's used the very next line, it would be nice to just re-use the `hash` variable